components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -598,14 +606,44 @@ void RewardsServiceImpl::CreateRewardsWallet(
         observer.OnRewardsWalletUpdated();
       }
 
-      std::move(callback).Run(ledger::mojom::Result::LEDGER_OK);
+      std::move(callback).Run(result);
     };
 
     self->bat_ledger_->CreateRewardsWallet(
-        base::BindOnce(on_created, self, std::move(callback)));
+        country,
+        base::BindOnce(on_created, self, country, std::move(callback)));
   };
 
-  StartProcess(base::BindOnce(on_start, AsWeakPtr(), std::move(callback)));
+  StartProcess(
+      base::BindOnce(on_start, AsWeakPtr(), country, std::move(callback)));
+}
+
+std::string RewardsServiceImpl::GetCountryCode() const {
+  auto* prefs = profile_->GetPrefs();
+  std::string country = prefs->GetString(prefs::kDeclaredGeo);
+  if (!country.empty()) {
+    return country;
+  }
+
+  int32_t country_id = country_id_;
+  if (!country_id) {
+    country_id = country_codes::GetCountryIDFromPrefs(prefs);
+  }
+  if (country_id < 0) {
+    return "";
+  }
+  return {base::ToUpperASCII(static_cast<char>(country_id >> 8)),
+          base::ToUpperASCII(static_cast<char>(0xFF & country_id))};
+}
+
+void RewardsServiceImpl::GetAvailableCountries(
+    GetAvailableCountriesCallback callback) const {
+  std::vector<std::string> countries;
+  for (const char* const* country_pointer = icu::Locale::getISOCountries();
```

Comment:
I'd probably cache this in a `static` local variable (`static const std::vector<std::string> countries = []{ /* ... */ }();`).

Reply:
Updated to use a static variable (also moved the list-building logic to a free function).